python_modules/libraries/dagster-dg/dagster_dg/cli/code_location.py

```diff
@@ -40,9 +41,12 @@ def code_location_group():
     default=False,
     help="Do not create a virtual environment for the code location.",
 )
-@click.pass_context
+@dg_global_options
 def code_location_generate_command(
-    cli_context: click.Context, name: str, use_editable_dagster: Optional[str], skip_venv: bool
+    name: str,
+    use_editable_dagster: Optional[str],
+    skip_venv: bool,
+    **global_options: object,
```

Comment:
can we type this with a `TypedDict`?

Reply:
I think the repetition with `DgConfig` is more trouble than its worth. We always process it immediately, and the use of `object` (as opposed to `Any`) means that it will cause a type error if we do anything to it without processing it.